### PR TITLE
Add .gitignore file for Hugo.

### PR DESCRIPTION
**Pull request by felicianotech**
**Reasons for making this change:**

Hugo is missing and has a directory, where files are generated, that should be ignored.

**Links to documentation supporting these rule changes:** 

https://gohugo.io/tutorials/automated-deployments#setting-up-version-control

If this is a new template: 

- **Link to application or project’s homepage**: https://gohugo.io/
- Hugo has grown greatly in the last few years. It's now [considered the # 2 most popular static site generator](https://www.netlify.com/blog/2017/05/25/top-ten-static-site-generators-of-2017/?utm_content=buffer4bc8a&utm_medium=social&utm_source=twitter.com&utm_campaign=buffer
) right behind Jekyll.

### CODE REVIEW

File: Hugo.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/public/
```
Comment (bep) [inline Hugo.gitignore]: I suspect the more correct would be "public/"

Comment (felicianotech) [inline Hugo.gitignore]: The tutorial link leads with a forward slash. *shrug*. Updated.

**Comment by joeblau** [comment]
@felicianotech this is included on https://github.com/dvcs/gitignore

https://www.gitignore.io/api/hugo

**Comment by shiftkey** [comment]
@felicianotech apologies for the delay in replying to this.

I had a lot of these templates requested, but I couldn't quite figure out the right place for them. I ended up settling on #2857, and you'll see the Hugo template can now be found at `community/Golang/Hugo.gitignore`.

Feel free to submit any future improvements to that template, and thanks again for the contribution!

File: Hugo.gitignore (hunk 1 above)
Comment (TamaMcGlinn) [reply Hugo.gitignore]: leading with a forward slash is the better approach. It ignores only the one directory called public in the root of the git repository. Without a leading slash, 'public/' ignores all directories that are called public, no matter where they are.

**Pull request closed without merging**